### ISSUE TITLE
jobs stopping (or going real slow) while spades running

### ISSUE THREAD

**Issue by angelajmcd** [mossmatters/HybPiper#37]
Hi Matt,

I recently got a new computer, so the first thing I did was reinstall my favorite software (hybpiper and dependencies, of course!).  When I run reads_first.py --check-depend, everything looks good to go. Then, when I run hybpiper for one sample on my data, it stalls when SPAdes is running. I double checked installations for everything and they seem ok. I am currently running the test dataset through, and the same thing is happening. It will sit for hours with no apparent progress being made. Do you think it could be the parallel jobs running? Can I run it with fewer cores? I never worried about specifying that before. I'll paste what it looks like below, though I think it looks normal...  it just sits like this for many hours, if I allow it.  :( 

Any tips? 
Angela

AngJMcD:test_dataset angelarein$ ../reads_first.py -b test_targets.fasta -r NZ281_R*_test.fastq --prefix NZ281 --bwa
HybPiper was called with these arguments:
../reads_first.py -b test_targets.fasta -r NZ281_R1_test.fastq NZ281_R2_test.fastq --prefix NZ281 --bwa

Making nucleotide bwa index in current directory.
[CMD]: bwa index test_targets.fasta
[bwa_index] Pack FASTA... 0.00 sec
[bwa_index] Construct BWT for the packed sequence...
[bwa_index] 0.01 seconds elapse.
[bwa_index] Update BWT... 0.00 sec
[bwa_index] Pack forward-only FASTA... 0.00 sec
[bwa_index] Construct SA from BWT and Occ... 0.00 sec
[main] Version: 0.7.15-r1140
[main] CMD: bwa index test_targets.fasta
[main] Real time: 0.013 sec; CPU: 0.017 sec
[CMD]: time bwa mem -t 8 test_targets.fasta /Applications/HybPiper/test_dataset/NZ281_R1_test.fastq /Applications/HybPiper/test_dataset/NZ281_R2_test.fastq  | samtools view -h -b -S - >  NZ281.bam
[M::bwa_idx_load_from_disk] read 0 ALT contigs
[M::process] read 22854 sequences (3849849 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 423, 0, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (151, 215, 298)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 592)
[M::mem_pestat] mean and std.dev: (227.95, 117.58)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 739)
[M::mem_pestat] skip orientation RF as there are not enough pairs
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_process_seqs] Processed 22854 reads in 2.376 CPU sec, 1.581 real sec
[main] Version: 0.7.15-r1140
[main] CMD: bwa mem -t 8 test_targets.fasta /Applications/HybPiper/test_dataset/NZ281_R1_test.fastq /Applications/HybPiper/test_dataset/NZ281_R2_test.fastq
[main] Real time: 4.672 sec; CPU: 2.436 sec

real	0m4.780s
user	0m2.994s
sys	0m0.035s
[CMD] time python /Applications/HybPiper/distribute_reads_to_targets_bwa.py NZ281.bam /Applications/HybPiper/test_dataset/NZ281_R1_test.fastq /Applications/HybPiper/test_dataset/NZ281_R2_test.fastq

Unique reads with hits: 1921

real	0m2.735s
user	0m0.404s
sys	0m0.259s
[DISTRIBUTE]: time python /Applications/HybPiper/distribute_targets.py /Applications/HybPiper/test_dataset/test_targets.fasta --bam NZ281.bam
/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/Bio/Seq.py:2071: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.
  BiopythonWarning)
[DISTRIBUTE]: 0 proteins had no good matches.

real	0m1.050s
user	0m0.205s
sys	0m0.078s
Running SPAdes on 13 genes
time parallel --eta spades.py --only-assembler --threads 1 --cov-cutoff 8 --12 {}/{}_interleaved.fasta -o {}/{}_spades :::: spades_genelist.txt > spades.log

Computers / CPU cores / Max jobs to run
1:local / 8 / 8

Computer:jobs running/jobs completed/%of started jobs/Average seconds to complete
ETA: 0s Left: 13 AVG: 0.00s  local:8/0/100%/0.0s

**Comment by angelajmcd** [comment]
Hi again,

When I run ./run_tests.sh instead of running one sample at a time, it looks like the genes_with_seqs.txt file is never made... here's the error I get at the end.
.
.
.

ERROR: No genes with BLAST hits! Exiting!
ERROR: cannot find genes_with_paralog_warnings.txt in EG30!
ERROR: cannot find genes_with_paralog_warnings.txt in EG98!
ERROR: cannot find genes_with_paralog_warnings.txt in MWL2!
ERROR: cannot find genes_with_paralog_warnings.txt in NZ281!
ERROR: cannot find genes_with_paralog_warnings.txt in NZ728!
ERROR: cannot find genes_with_paralog_warnings.txt in NZ739!
ERROR: cannot find genes_with_paralog_warnings.txt in NZ866!
ERROR: cannot find genes_with_paralog_warnings.txt in NZ874!
ERROR: cannot find genes_with_paralog_warnings.txt in NZ911!
Traceback (most recent call last):
  File "../intronerate.py", line 307, in <module>
    if __name__ == "__main__":main()
  File "../intronerate.py", line 268, in main
    genelist = [x.split()[0] for x in open('genes_with_seqs.txt').readlines()]
IOError: [Errno 2] No such file or directory: 'genes_with_seqs.txt'
Traceback (most recent call last):
  File "../intronerate.py", line 307, in <module>
    if __name__ == "__main__":main()
  File "../intronerate.py", line 268, in main
    genelist = [x.split()[0] for x in open('genes_with_seqs.txt').readlines()]
IOError: [Errno 2] No such file or directory: 'genes_with_seqs.txt'
Traceback (most recent call last):
  File "../intronerate.py", line 307, in <module>
    if __name__ == "__main__":main()
  File "../intronerate.py", line 268, in main
    genelist = [x.split()[0] for x in open('genes_with_seqs.txt').readlines()]
IOError: [Errno 2] No such file or directory: 'genes_with_seqs.txt'
Traceback (most recent call last):
  File "../intronerate.py", line 307, in <module>
    if __name__ == "__main__":main()
  File "../intronerate.py", line 268, in main
    genelist = [x.split()[0] for x in open('genes_with_seqs.txt').readlines()]
IOError: [Errno 2] No such file or directory: 'genes_with_seqs.txt'
Traceback (most recent call last):
  File "../intronerate.py", line 307, in <module>
    if __name__ == "__main__":main()
  File "../intronerate.py", line 268, in main
    genelist = [x.split()[0] for x in open('genes_with_seqs.txt').readlines()]
IOError: [Errno 2] No such file or directory: 'genes_with_seqs.txt'
Traceback (most recent call last):
  File "../intronerate.py", line 307, in <module>
    if __name__ == "__main__":main()
  File "../intronerate.py", line 268, in main
    genelist = [x.split()[0] for x in open('genes_with_seqs.txt').readlines()]
IOError: [Errno 2] No such file or directory: 'genes_with_seqs.txt'
Traceback (most recent call last):
  File "../intronerate.py", line 307, in <module>
    if __name__ == "__main__":main()
  File "../intronerate.py", line 268, in main
    genelist = [x.split()[0] for x in open('genes_with_seqs.txt').readlines()]
IOError: [Errno 2] No such file or directory: 'genes_with_seqs.txt'
Traceback (most recent call last):
  File "../intronerate.py", line 307, in <module>
    if __name__ == "__main__":main()
  File "../intronerate.py", line 268, in main
    genelist = [x.split()[0] for x in open('genes_with_seqs.txt').readlines()]
IOError: [Errno 2] No such file or directory: 'genes_with_seqs.txt'
Traceback (most recent call last):
  File "../intronerate.py", line 307, in <module>
    if __name__ == "__main__":main()
  File "../intronerate.py", line 268, in main
    genelist = [x.split()[0] for x in open('genes_with_seqs.txt').readlines()]
IOError: [Errno 2] No such file or directory: 'genes_with_seqs.txt'

**Comment by mossmatters** [comment]
Lots of people seem to be having issues with the SPAdes phase recently, and the tip that you've got it happening on a new computer should help me a lot!

As for the `run_tests.sh` problem, the issue is in the shell script. You need to remove the quotes from `"$i*.fasta"`. I've learned the wild card * does not expand properly inside a shell script, which makes HybPiper think you've got single-end data, in turn resulting in the "No BLAST hits found" error. 

I will be sure to remove those quotes for future versions!

**Comment by angelajmcd** [comment]
HI, so I may have figured it out.... 

The SPAdes I installed through homebrew/brewsci is the problem, I think. That install gave this error, which when I googled it, seemed mostly inert according to the google gurus:

==> Pouring spades-3.11.1_2.high_sierra.bottle.tar.gz
Warning: spades dependency gmp was built with a different C++ standard
library (libc++ from clang). This may cause problems at runtime.

I guess it isn't inert. I manually installed the newest spades and added it to my path, and HybPiper is working fast now. 

Yes, I figured the quotes thing out. It was throwing errors even without them because of the spades deal. I just pasted the sillier one, haha.

Hope you're well!
